coalib/results/Result.py

```diff
@@ -63,6 +65,13 @@ def __init__(self,
         if severity not in RESULT_SEVERITY.reverse:
             raise ValueError("severity is not a valid RESULT_SEVERITY")
 
+        self.diffs = {}
+        if(diffs and type(diffs) != dict):
+            for diff in diffs:
+                self.diffs[diff.filename] = diff
```

Comment:
use a generator expression here, much faster in terms of performance and usually simpler to understand